biofefi/services/fi.py

```diff
@@ -78,3 +92,66 @@ def calculate_local_shap_values(
 
     # Return the DataFrame
     return shap_df, shap_values
+
+
+def load_fi_options(path: Path) -> dict:
+    """Load feature importance options.
+
+    Args:
+        path (Path): The path to the feature importance options file.
+
+    Returns:
+        dict: The feature importance options.
+    """
+
+    try:
+        with open(path, "r") as file:
+            fi_options = json.load(file)
+    except FileNotFoundError:
+        fi_options = None
+
+    return fi_options
+
+
+def find_previous_fi_results(experiment_path: Path) -> tuple[dict, dict]:
+    """Find previous feature importance results.
+
+    Args:
+        experiment_path (Path): The path to the experiment.
+
+    Returns:
+        tuple[dict, dict]: The feature importance options and results.
+    """
+    try:
+        fi_options = load_fi_options(fi_options_path(experiment_path))
+
+    except Exception:
+        fi_options = None
+
+    return fi_options
+
+
+def delete_previous_FI_results(experiment_path: Path):
+    """Delete previous feature importance results.
+
+    Args:
+        experiment_path (Path): The path to the experiment.
+    """
+    if os.path.exists(fi_plot_dir(experiment_path)):
+        delete_directory(fi_plot_dir(experiment_path))
+    if os.path.exists(fi_result_dir(experiment_path)):
+        delete_directory(fi_result_dir(experiment_path))
+    if os.path.exists(fi_options_dir(experiment_path)):
+        delete_directory(fi_options_dir(experiment_path))
+    if os.path.exists(fuzzy_plot_dir(experiment_path)):
+        delete_directory(fuzzy_plot_dir(experiment_path))
+    if os.path.exists(fuzzy_result_dir(experiment_path)):
+        delete_directory(fuzzy_result_dir(experiment_path))
+    if os.path.exists(fuzzy_options_path(experiment_path)):
+        delete_directory(fuzzy_options_path(experiment_path))
+    if os.path.exists(fi_options_path(experiment_path)):
+        delete_directory(fi_options_path(experiment_path))
+    if os.path.exists(log_dir(experiment_path) / "fi"):
+        delete_directory(log_dir(experiment_path) / "fi")
+    if os.path.exists(log_dir(experiment_path) / "fuzzy"):
+        delete_directory(log_dir(experiment_path) / "fuzzy")
```

Comment:
This is very repetitive. What you can do instead is initialise the paths is variables inside a list. Then iterate over the list checking if each path exists and delete them if the do. Also the return types of those file path functions should all be `Path`, so they will have a `.exists()` method to use instead of `os.path.exists`. So it'll be more concise.

Reply:
Same here, move to `services.experiments`